From ab5d6985ebb1ca20c6815ebf8b400180915f2c5c Mon Sep 17 00:00:00 2001
From: Srinivas Neeli <srinivas.neeli@xilinx.com>
Date: Tue, 15 Sep 2020 20:36:18 +0530
Subject: [PATCH 1589/1851] watchdog: of_xilinx_wdt: Add interrupt and
 pretimeout properties for Versal watchdog

commit 51e28426213c6350d23c4fb4780241d12ecccc0d from
https://github.com/Xilinx/linux-xlnx.git

Updated interrupt and pretimeout-sec properties for versal window watchdog.

Signed-off-by: Srinivas Neeli <srinivas.neeli@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 .../devicetree/bindings/watchdog/of-xilinx-wdt.txt   | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/Documentation/devicetree/bindings/watchdog/of-xilinx-wdt.txt b/Documentation/devicetree/bindings/watchdog/of-xilinx-wdt.txt
index 10d68003158d..57903645d685 100644
--- a/Documentation/devicetree/bindings/watchdog/of-xilinx-wdt.txt
+++ b/Documentation/devicetree/bindings/watchdog/of-xilinx-wdt.txt
@@ -20,6 +20,15 @@ Optional properties for AXI/PLB soft-core watchdog:
 
 Optional properties for window watchdog:
 - timeout-sec          : Watchdog timeout value (in seconds).
+			 if unset, the default timeout is 10 seconds.
+- pretimeout-sec	: Watchdog pretimeout value in seconds.
+- interrupts		: IRQ line for the WWDT.
+- interrupt-names	: Interrupt line names "wdt" or "wwdt_reset_pending".
+			  wdt - will assert high after first (closed) window
+			  timer expires. wwdt_reset_pending - will assert high
+			  after second (open) window timer expires if WRP
+			  (Watchdog Reset pending) is configured with third
+			  (SST) timer.
 
 Example:
 Xilinx AXI/PLB soft-core watchdog:
@@ -38,4 +47,7 @@ watchdog@fd4d0000 {
 	reg = <0x0 0xfd4d0000 0x0 0x10000>;
 	clocks = <&clk25>;
 	timeout-sec = <10>;
+	interrupt-names = "wdt", "wwdt_reset_pending";
+	interrupts = <0x0 0x64 0x1>, <0x0 0x6D 0x1>;
+	pretimeout-sec = <5>;
 } ;
-- 
2.31.1

